common/performance.py

```
@@ -0,0 +1,720 @@
+        logger.error(f"Failed to clear cache: {e}")
+
+
+def get_cache_stats() -> Dict[str, Any]:
+    """Get comprehensive cache statistics."""
+    try:
+        # Get stats from the generalized framework
+        performance_summary = get_cache_performance_summary()
+
+        # Add legacy cache stats if available
+        legacy_stats = None
+        if hasattr(cache, "_cache") and hasattr(cache._cache, "get_stats"):
```

Pylint (3.3.7)

[warning] Access to a protected member _cache of a client class

(W0212)